jobs/bodhi-trigger.Jenkinsfile

```diff
@@ -0,0 +1,202 @@
+                [field: "\$.update.builds[?(@.type == 'rpm' && @.nvr =~ /^(${srpms.join("|")})-[^-]+-[^-]+\$/)]", expectedValue: '^\\[.+\\]$']
+            ]
+        ),
+        // TL;DR: Effectively triggered when a non-rawhide update was opened.
+        // Longer explanation: Triggered when an update was requested to be
+        // pushed to updates-testing. This never happens for rawhide, and for
+        // non-rawhide usually happens shortly after the update is opened.
+        rabbitMQSubscriber(
+            name: 'Fedora Messaging',
+            overrides: [
+                topic: 'org.fedoraproject.prod.bodhi.update.request.testing',
+                queue: '6455c3a4-9194-4cc0-a7f6-79864036ded2'
+            ],
+            checks: [
+                [field: '$.update.release.id_prefix', expectedValue: '^FEDORA$'],
+                [field: '$.update.release.name', expectedValue: "^F(${releasevers.values().join("|")})\$"],
+                [field: "\$.update.builds[?(@.type == 'rpm' && @.nvr =~ /^(${srpms.join("|")})-[^-]+-[^-]+\$/)]", expectedValue: '^\\[.+\\]$']
+            ]
+        ),
+        // Triggered when an update's tests are requested to be rerun.
+        rabbitMQSubscriber(
+            name: 'Fedora Messaging',
+            overrides: [
+                topic: 'org.fedoraproject.prod.bodhi.update.status.testing.koji-build-group.build.complete',
+                queue: '036407bc-20ad-4290-bdee-deeab6ff6d7a'
+            ],
+            checks: [
+                [field: '$.update.release.id_prefix', expectedValue: '^FEDORA$'],
+                [field: '$.update.release.name', expectedValue: "^F(${releasevers.values().join("|")})\$"],
+                [field: "\$.update.builds[?(@.type == 'rpm' && @.nvr =~ /^(${srpms.join("|")})-[^-]+-[^-]+\$/)]", expectedValue: '^\\[.+\\]$'],
+                // Were we asked to re-run the tests?
+                [field: '$.re-trigger', expectedValue: '^true$']
+            ]
+        ),
+        // Triggered when an update is edited. We only care about the case when
+        // the edit changes the builds in the update (and not e.g. changing
+        // the description).
+        rabbitMQSubscriber(
+            name: 'Fedora Messaging',
+            overrides: [
+                topic: 'org.fedoraproject.prod.bodhi.update.edit',
+                queue: '9a5c5403-0d3f-403c-81b0-e0a6414513b2'
+            ],
+            checks: [
+                [field: '$.update.release.id_prefix', expectedValue: '^FEDORA$'],
+                [field: '$.update.release.name', expectedValue: "^F(${releasevers.values().join("|")})\$"],
+                [field: "\$.update.builds[?(@.type == 'rpm' && @.nvr =~ /^(${srpms.join("|")})-[^-]+-[^-]+\$/)]", expectedValue: '^\\[.+\\]$'],
+                // Were any builds actually changed?
+                [field: 'concat($.new_builds,$.removed_builds)', expectedValue: '^.*$'],
+            ]
+        )
+      ]
```

Comment:
/cc @AdamWill

If you have a chance, would appreciate a sanity-check on the various conditions here.